snuba/utils/streams/batching.py

```diff
@@ -99,6 +102,17 @@ def __init__(
         max_batch_time: int,
         metrics: MetricsBackend,
         recoverable_errors: Optional[Sequence[Type[ConsumerError]]] = None,
+        commit_retry_policy: RetryPolicy = BasicRetryPolicy(
+            4,
+            constant_delay(1),
+            lambda e: isinstance(e, KafkaException)
+            and e.args[0].code()
+            in (
+                KafkaError.REQUEST_TIMED_OUT,
+                KafkaError.NOT_COORDINATOR_FOR_GROUP,
+                KafkaError._WAIT_COORD,
+            ),
+        ),
```

Comment:
I think it would be a bit risky to provide the instance as default constructor parameter. Since the instance of commit_retry_policy would be the same for any instance of BatchingConsumer we would have  to remember about this should we add any mutable state to BasicRetryPolicy. Unless you need everyone to share the same instance, then it seems risky on the other way.

Reply:
Fair point, although I don't think the retry policies should be mutable. That should probably be noted in their documentation.

In any case, I don't think this is the right home for this (I noted this in Slack but probably should have commented that here instead) since it's coupled to the Kafka implementation, and the batching consumer generically is not.

I think this should be passed in via the constructor (and it should default to `NoRetryPolicy`) but am not sure where it should be instantiated — it probably should be derived from configuration somewhere, and eventually as part of [`ConsumerBuilder`](https://github.com/getsentry/snuba/blob/988a4312fc9c107bc735fb2295e269b01ef2dea4/snuba/consumers/consumer_builder.py#L14)… I guess it could even be defined statically there for now until there's a scenario that requires it to be configurable?